big-red-shuttle/ScheduleViewController.swift

```diff
@@ -0,0 +1,120 @@
+        //Bar
+        scheduleBar = ScheduleBar(frame: CGRect(x: 0, y: 0, width: self.view.bounds.width, height: barHeight))
+        scheduleBar.setUp(buttonsData: looptimes, selected: 2)
+        scheduleBar.delegateSB = self
+        self.view.addSubview(scheduleBar)
+        
+        //Table
+        table = UITableView(frame: CGRect(x: 0, y: scheduleBar.frame.maxY, width: view.bounds.width, height: view.bounds.height - (navigationController?.navigationBar.bounds.height)! - (navigationController?.tabBarController?.tabBar.bounds.height)! - scheduleBar.bounds.height))
+        table.separatorStyle = UITableViewCellSeparatorStyle.none
+        table.delegate = self
+        table.dataSource = self
+        self.view.addSubview(table)
+        
+        //Nav
+        self.navigationController?.navigationBar.topItem?.title = "Schedule"
+        
+    }
+    
+    override func viewWillAppear(_ animated: Bool){
+        super.viewWillAppear(animated)
+        table.register(ScheduleTableViewCell.classForCoder(), forCellReuseIdentifier: identifier)
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    
+    // MARK: - Schedule Bar delegate
+    
+    func selectCell(button: UIButton) {
+        let time = looptimes[button.tag]
+        times.index(of: time)
+        let indexPath = IndexPath(row: button.tag, section: 0)
+        table.scrollToRow(at: indexPath, at: UITableViewScrollPosition.top, animated: true)
+    }
+    
+    
+    // MARK: - Table view data source
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return times.count
+    }
+    
+    // MARK: - Table view delegate
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        var cell = tableView.dequeueReusableCell(withIdentifier: identifier, for: indexPath) as? ScheduleTableViewCell
+        if cell == nil{
+            cell = ScheduleTableViewCell(style: UITableViewCellStyle.default, reuseIdentifier: identifier)
+        }
+        
+        var loopBeginning: Bool
+        if stops[indexPath.row] == "Jessup & Program House Drive"{
+            loopBeginning = true
+        }else{
+            loopBeginning = false
+        }
+        cell?.configLoopBegin(loop: loopBeginning)
+        cell?.time.text = times[indexPath.row]
+        cell?.stop.text = stops[indexPath.row]
+        cell?.time.sizeToFit()
+        cell?.stop.sizeToFit()
+        
+        return cell!
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return cellHeight
+    }
```

Comment:
Set the property on UITableView directly and don't implement this method! You take a performance hit if you implement this without needing it.